Merge the final LoRA finetuned checkpoint (#1081)
Lightning-AI/litgpt: tutorials/finetune_lora.md

&nbsp;

## Merging LoRA Weights (Optional)

Finetuning a model with LoRA generates a `lit_model.pth.lora` file.

> This file exclusively contains the LoRA weights, which has is much smaller than the original model checkpoint to conserve storage space.

Reviewer: ```suggestion
This file exclusively contains the LoRA weights, which are much smaller than the original model checkpoint to conserve storage space.
```

If desired, there is the option to merge these LoRA weights directly into the original model's checkpoint, which creates a full `lit_model.pth` checkpoint.
The advantage of this merging process is to streamline inference operations, as it eliminates the need to dynamically incorporate the LoRA weights during runtime, which can improve inference speed.

> > [!NOTE]

Author: Thanks, I was looking for a good spot for this

> LitGPT will automatically merge the checkpoint for you if you use it in any of the inference commands, such as `litgpt generate` or `litgpt chat`.
> Manual merging is only necessary if you want to use the checkpoint outside LitGPT.

> For example, after finetuning produced a checkpoint folder `out/lora/step-002000`, merge it like so:

Reviewer: ```suggestion
For example, after finetuning produced a checkpoint folder `out/lora/step-002000`, merge it as follows:
```

```bash
litgpt merge_lora --checkpoint_dir "out/lora/step-002000"
```

> Executing this results in the creation of a full `lit_model.pth` checkpoint file.

Reviewer: ```suggestion
The command above creates a full `lit_model.pth` checkpoint file.
```
